internal/cmd/configssh.go

```diff
@@ -240,32 +242,41 @@ func makeNewConfigs(binPath string, workspaces []coderutil.WorkspaceWithWorkspac
 			continue
 		}
 
-		newConfig += makeSSHConfig(binPath, workspace.Workspace.Name, privateKeyFilepath)
+		newConfig += makeSSHConfig(binPath, workspace.Workspace.Name, privateKeyFilepath, additionalOptions)
 	}
 	newConfig += fmt.Sprintf("\n%s\n", sshEndToken)
 
 	return newConfig
 }
 
-func makeSSHConfig(binPath, workspaceName, privateKeyFilepath string) string {
-	entry := fmt.Sprintf(
-		`Host coder.%s
-   HostName coder.%s
-   ProxyCommand "%s" tunnel %s 12213 stdio
-   StrictHostKeyChecking no
-   ConnectTimeout=0
-   IdentitiesOnly yes
-   IdentityFile="%s"
-`, workspaceName, workspaceName, binPath, workspaceName, privateKeyFilepath)
+func makeSSHConfig(binPath, workspaceName, privateKeyFilepath string, additionalOptions []string) string {
+	// Custom user options come first to maximizessh customization.
+	options := []string{}
+	if len(additionalOptions) > 0 {
+		options = []string{
+			"# Custom options. Duplicated values will always prefer the first!",
```

Comment:
I think it would be better if we kept a `map[string]string` of the name/value pairs, and then override them based on key... but ssh might have some peculiarities with the config format, like being case-insensitive

I think this is an improvement over the status quo so definitely think this is good now, we can always revisit later if we get customer feedback that this is confusing :)

Reply:
I did this in an iteration, but didn't want the jank of splitting on " " or "="... I wasn't sure if there were additional edge-cases, and didn't wanna risk it!

That's definitely a cleaner approach though, so someday :pray: